### PR TITLE
Fixes NPE when receipt takes longer than expected time

### PR DESCRIPTION
**Pull request by otherview**
### Why is this change needed?

- As per title, a return was needed otherwise `receipt` would create a NPE at the trace operation

### What changes were made as part of this PR:

- Provide a high level list of the changes made
- List key areas for the reviewer 

### :rotating_light: Definition of done :rotating_light:
- [ ] Unit tests added to cover new or changed functionality 
- [ ] Docs pages updated to cover new or changed functionality
- [ ] [Changelog.md](https://github.com/obscuronet/go-obscuro/blob/main/docs/testnet/changelog.md) updated

### CODE REVIEW

File: go/host/host.go [hunk 1]
```diff
@@ -650,15 +650,20 @@ func (h *host) signAndBroadcastL1Tx(tx types.TxData, tries uint64) error {
 func (h *host) watchForReceipt(txHash common.TxHash) {
 	var receipt *types.Receipt
 	var err error
-	err = retry.Do(func() error {
-		receipt, err = h.ethClient.TransactionReceipt(txHash)
-		return err
-	}, retry.NewTimeoutStrategy(maxWaitForL1Receipt, retryIntervalForL1Receipt))
+	err = retry.Do(
```
Comment (BedrockSquirrel) [inline go/host/host.go:653]: Minor (subjective) thing but I preferred the formatting as it was before tbh.

When I see functions as parameters in Go code it usually seems to be like that, with the anonymous func started inline to reduce the indentation, like javascript-style (e.g. the `sub.Receive()` example in the [pubsub callback Go docs](https://pkg.go.dev/cloud.google.com/go/pubsub)).

**Review by BedrockSquirrel — APPROVED** [review]
LGTM, thanks for fixing! 😳

File: go/host/host.go (hunk 1 above)
Comment (BedrockSquirrel) [reply go/host/host.go:653]: Eh, looking around a bit more I see both styles haha. It's probably just what I'm used to from JS.

Comment (otherview) [reply go/host/host.go:653]: I typically put one thing per line.. I think this is a job for linter :)

**Pull request merged**